robot/server/server.go

```diff
@@ -231,6 +233,34 @@ func (s *Server) TransformPose(ctx context.Context, req *pb.TransformPoseRequest
 	return &pb.TransformPoseResponse{Pose: referenceframe.PoseInFrameToProtobuf(transformedPose)}, err
 }
 
+// TransformPCD will transform the pointcloud to the desired frame in the robot's frame system.
+// Do not move the robot between the generation of the initial pointcloud and the receipt
+// of the transformed pointcloud because that will make the transformations inaccurate.
+// TODO(RSDK-1123): PCD files have a field called VIEWPOINT which encodes an offset as a translation+quaternion.
+// if we used VIEWPOINT, you only need to query the frame system to get the transform between the source and destination frame.
+// Then, you put that transform as a translation+quaternion in the VIEWPOINT field. You would only change one line in the PCD file,
+// rather than having to decode and then encode every point in the PCD. Would be a considerable speed up.
+func (s *Server) TransformPCD(ctx context.Context, req *pb.TransformPCDRequest) (*pb.TransformPCDResponse, error) {
+	// transform PCD bytes to pointcloud
+	pc, err := pointcloud.ReadPCD(bytes.NewReader(req.PointCloudPcd))
+	if err != nil {
+		return nil, err
+	}
+	// transform
+	final, err := s.r.TransformPointCloud(ctx, pc, req.Source, req.Destination)
+	if err != nil {
+		return nil, err
+	}
+	// transform pointcloud back to PCD bytes
+	var buf bytes.Buffer
+	buf.Grow(200 + (final.Size() * 4 * 4)) // 4 numbers per point, each 4 bytes
+	err = pointcloud.ToPCD(final, &buf, pointcloud.PCDBinary)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This part can be removed by implementing RSDK-1123 and would be a considerable speed-up. Then it would just a be a matter of finding the transform pose from the frame system, editing the bytes for the VIEWPOINT to encode the transform, and then sending along the edited PCD.